target/linux/ar71xx/patches-4.4/703-MIPS-ath79-fix-CPE210-v2-refclk.patch

```diff
@@ -0,0 +1,60 @@
+From: Sven Eckelmann <sven@nafation.org>
+Date: Wed, 17 Jan 2018 19:41:52 +0100
+Subject: [PATCH] MIPS: ath79: fix CPE210 v2 refclk
+
+The TP-Link CPE210 v2 uses a 25 MHz reference clock on the board. But the
+bootstrap register is setting the device up as a 40 MHz refclk based
+board design.
+
+Unfortunately, the qca953x_clocks_init is started before the board specific
+device code is started. It must therefore be hardcoded in the actual init
+function.
+
+Signed-off-by: Sven Eckelmann <sven@narfation.org>
+---
+ arch/mips/ath79/clock.c    | 6 +++++-
+ arch/mips/ath79/dev-wmac.c | 7 +++----
+ 2 files changed, 8 insertions(+), 5 deletions(-)
+
+--- a/arch/mips/ath79/clock.c
++++ b/arch/mips/ath79/clock.c
+@@ -24,6 +24,7 @@
+ #include <asm/mach-ath79/ath79.h>
+ #include <asm/mach-ath79/ar71xx_regs.h>
+ #include "common.h"
++#include "machtypes.h"
+ 
+ #define AR71XX_BASE_FREQ	40000000
+ #define AR724X_BASE_FREQ	40000000
+@@ -365,7 +366,9 @@ static void __init qca953x_clocks_init(v
+ 	u32 bootstrap;
+ 
+ 	bootstrap = ath79_reset_rr(QCA953X_RESET_REG_BOOTSTRAP);
+-	if (bootstrap &	QCA953X_BOOTSTRAP_REF_CLK_40)
++	if (mips_machtype == ATH79_MACH_CPE210_V2)
```

Comment:
Personally, I think that using here a statement based on mach type might make things more complicated in (near) future. I wouldn't be surprised to see more TP-Link boards coming with the same approach which would require updating this patch over and over again.

Based on all QCA953x datasheets which [are available on-line](https://github.com/Deoptim/atheros/), this SoC series just doesn't support other reference clock type than 25 MHz and I haven't seen board with different clock so far.

![qca953x-bootstrap](https://user-images.githubusercontent.com/2934612/35148198-a5e6ab52-fd11-11e7-9f5d-dbe4068a718c.png)

I'm not going to block this PR because of these two patches but I think that it would be safe to just set a static 25 MHz reference clock for QCA953x SoC series.

@ecsv, as you are the author of the patches, what is your opinion?